src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -14,171 +14,177 @@ import AlertDialogBox from '../../../../components/AlertDialogBox';
-    this.offCampusProgramInfo = new Map();
-    this.state.applicants.forEach((profile) =>
-      this.offCampusProgramInfo.set(profile.AD_Username, ''),
-    );
-    this.editDialogText = '';
-    this.snackbarText = '';
-    this.snackbarSeverity = '';
-    this.saveButtonAlertTimeout = null;
-  }
-
-  componentDidMount() {
-    this.loadProfile();
-    this.loadSavedApplication();
-  }
+const InstructionsCard = () => (
+  <Card>
+    <CardHeader title="Apartment Application Instructions" className="card-header" />
+    <CardContent>
+      <Typography variant="body1">Placeholder Text</Typography>
+      <Typography variant="body1">
+        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos blanditiis tenetur unde
+        suscipit, quam beatae rerum inventore consectetur, neque doloribus, cupiditate numquam
+        dignissimos laborum fugiat deleniti? Eum quasi quidem quibusdam.
+      </Typography>
+    </CardContent>
+  </Card>
+);
+
+const StudentApplication = (props) => {
+  const [loading, setLoading] = useState(true);
+  const [saving, setSaving] = useState(false);
+
+  const [applicationID, setApplicationID] = useState(-1); // Default value of -1 indicate to backend that the application ID number is not yet known
+  const [editorUsername, setEditorUsername] = useState(null); // The username of the application editor
+  const [applicants, setApplicants] = useState([]);
+  const [preferredHalls, setPreferredHalls] = useState([{ HallRank: 1, HallName: '' }]);
+  // const [offCampusProgramInfo, setOffCampusProgramInfo] = useState(new Map());
+
+  const [applicationCardsOpen, setApplicationCardsOpen] = useState(false);
+  const [newEditorProfile, setNewEditorProfile] = useState(null); // Stores the StudentProfileInfo of the new editor before the user confirms the change
+  const [changeEditorDialogOpen, setChangeEditorDialogOpen] = useState(false);
+  const [submitDialogOpen, setSubmitDialogOpen] = useState(false); // Use this for submitting app (later feature)
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [snackbarText, setSnackbarText] = useState('');
+  const [snackbarSeverity, setSnackbarSeverity] = useState('');
+  const [saveButtonAlertTimeout, setSaveButtonAlertTimeout] = useState(null);
+
+  // const peopleSearchRef = useRef();
+
+  let offCampusProgramInfo = new Map();
+  applicants.forEach((profile) => offCampusProgramInfo.set(profile.AD_Username, ''));
 
   /**
-   * Loads the user's profile info only once (at start)
+   * Attempt to load an existing application from the database if one exists
    */
-  async loadProfile() {
-    this.setState({ loading: true });
-    try {
-      let applicants = this.state.applicants;
-      applicants.push(this.props.userProfile);
-      this.setState({ applicants });
-      this.setState({ loading: false });
-    } catch (error) {
-      // Do Nothing
-    }
-  }
+  useEffect(() => {
+    loadSavedApplication();
+  });
```

Comment:
This is probably meant to be called only on the first render, right? If so, you want to pass it an empty dependency array.